include/systems/system.h

```diff
@@ -1737,6 +1739,51 @@ class System : public ReferenceCountedObject<System>,
   void projection_matrix (SparseMatrix<Number> & proj_mat) const;
 #endif // LIBMESH_HAVE_METAPHYSICL
 
+  /**
+   * Adds an auxiliary matrix to the system. Must be implemented in derived classes
+   */
+  virtual Matrix & add_matrix(const std::string &,
+                              ParallelType = PARALLEL,
+                              MatrixBuildType = MatrixBuildType::AUTOMATIC)
+    {
+      libmesh_error_msg("The current system object does not support matrix addition");
+    }
+
+  /**
+   * Check whether a matrix exists in the system
+   *
+   * Must be implemented in derived classes
+   */
+  virtual bool have_matrix(const std::string &) const
+    {
+      libmesh_error_msg("The current system object does not support matrix addition");
```

Comment:
This would be the type of thing I'd prefer to avoid... I know we have base-class-error implementations elsewhere in the library, but it's kind of a hint that something is designed wrong when you have to do this IMO.